src/Monitoring/AutoWatcher.coffee

```diff
@@ -0,0 +1,89 @@
+AutoWatcher =
+  init: ->
+    return unless Conf['Filter']
+
+    AutoWatcher.periodicScan()
+
+  periodicScan: ->
+    clearTimeout AutoWatcher.timeout
+
+    interval = 5 * $.MINUTE
+
+    now = Date.now()
+
+    unless Conf['autoWatchLastScan'] and (now - interval < Conf['autoWatchLastScan'])
+      AutoWatcher.scan()
+      $.set 'autoWatchLastScan', now
+    AutoWatcher.timeout = setTimeout AutoWatcher.periodicScan, interval
+
+  scan: ->
+    sitesAndBoards = (for own _, filters of Filter.filters
+      Object.keys(filter.boards) for filter in filters when filter.watch and filter.boards
+    ).flat(2).reduce (acc, i) ->
+      [_, k, v] = i.match(/(.*)\/(.*)/)
+      acc[k] ?= []
+      acc[k].push(v)
+      acc
+    , {}
+    for own rawSite, boards of sitesAndBoards
+      break unless site = g.sites[rawSite]
+      for boardID in boards
+        AutoWatcher.fetchCatalog(boardID, site, AutoWatcher.parseCatalog)
+
+  fetchCatalog: (boardID, site, cb) ->
+    return unless url = site.urls['catalogJSON']?({boardID})
+
+    ajax = if site.ID is g.SITE.ID then $.ajax else CrossOrigin.ajax
+
+    onLoadEnd = ->
+      cb.apply @, [site, boardID]
+
+    $.whenModified(
+      url,
+      'AutoWatcher'
+      onLoadEnd,
+      {timeout: $.MINUTE, ajax}
+    )
```

Comment:
I'm wondering if it's possible to combine these periodic fetches with ThreadWatcher's periodic fetches so that less fetches have to take place, but that might be too much complexity.